README.md

```diff
@@ -241,30 +241,54 @@ If you set `exposeRoute` to `true` the plugin will expose the documentation with
 | `'/documentation/*'`    | External files that you may use in `$ref`  |
 
 <a name="register.options.transform"></a>
-#### Transforms
+#### Transform
 
-To use different schemas such as [Joi](https://github.com/hapijs/joi) you can pass a synchronous `transform` method in the options to convert them back to standard JSON schemas expected by this plugin to generate the documentation (`dynamic` mode only).
+By passing a synchronous `transform` function you can modify the route's url and schema.
+
+Some possible uses of this are:
+
+- add the `hide` flag on schema according to your own logic based on url & schema
+- altering the route url into something that's more suitable for the api spec
+- using different schemas such as [Joi](https://github.com/hapijs/joi) and transforming them to standard JSON schemas expected by this plugin
+
+This option is available in `dynamic` mode only.
+
+Examples of all the possible uses mentioned:
 
 ```js
 const convert = require('joi-to-json')
 
 fastify.register(require('fastify-swagger'), {
   swagger: { ... },
-  ...
-  transform: schema => {
+  transform: ({ schema, url }) => {
     const {
       params = undefined,
       body = undefined,
       querystring = undefined,
-      ...others
+      headers = undefined,
+      response = undefined,
+      ...rest
     } = schema
-    const transformed = { ...others }
-    if (params) transformed.params = convert(params)
-    if (body) transformed.body = convert(body)
-    if (querystring) transformed.querystring = convert(querystring)
-    return transformed
+    const transformedSchema = { ...rest }
```

Comment:
```suggestion
    const transformedSchema = Object.assign({}, rest)
```
Replacing rest by something better suited (see my previous comment).

Reply:
> why the Object.assign on target empty object? which is essesntialy a shallow copy over if i understand it correctly.

Here `Object.assign({}, rest)` is equivalent to `{ ...rest }` (it is just more explicit) ^^

> we're not mutating anything anywhere, should be ok to just use the original ...rest, or i'm missing something?

By essence `transform` will mutate the schema (and it's what it does in this example), this is why we initialize a new object instead of just doing something like this:

```js
// ...
const { params, body, querystring, headers, response, ...transformedSchema } = schema
let transformedUrl = url

if (params) transformedSchema.params = convert(params)
// ...
```

Reply:
hmm, this makes me think of 2 things:

1) if we want to ensure no mutation of existing `route.schema` which was passed in, then probably it'd be better to pass `lodash.cloneDeep(route.schema)` to the transform fn.

2)  returned schema should maybe also be some subtype of fastify schema.
to avoid users returning just part of the schema by mistake (without the `rest`).
```
transform?: <S extends FastifySchema = FastifySchema, T extends FastifySchema = FastifySchema>({schema, url}: {schema: S, url: string}) => { schema: T, url: string };
```

WDYT?

Reply:
No need to add a new dependency.
It's up to the user to decide if he wants to mutate the original schema or not (`transform` is about letting users apply the mutations of their choice after all 😄). In the documentation we just need to promote good practices IMO and by creating a new object with either `{ ...rest }` or `Object.assign({}, rest)` we do so and it is enough. 😛 

And about the second part, we expect a json schema as the output so checking that it is a json Object is enough and once again it's up to the user to make sure his transform function works as it should 😜